src/docfx/build/moniker/MonikerProvider.cs

```diff
@@ -81,36 +83,49 @@ public SourceInfo<string> GetConfigMonikerRange(FilePath file)
             return (null, monikers);
         }
 
-        private (Error? error, string[] monikers) GetFileLevelMonikersCore(FilePath file)
+        private (List<Error> errors, string[] monikers) GetFileLevelMonikersCore(FilePath file)
         {
+            var errors = new List<Error>();
             var (_, metadata) = _metadataProvider.GetMetadata(file);
 
             var configMonikerRange = GetConfigMonikerRange(file);
-            var configMonikers = _rangeParser.Parse(configMonikerRange);
+            var configedMonikers = _rangeParser.Parse(configMonikerRange);
 
-            if (!string.IsNullOrEmpty(metadata.MonikerRange))
+            if (!string.IsNullOrEmpty(metadata.MonikerRange) || metadata.Monikers.Length > 0)
             {
                 // Moniker range not defined in docfx.yml/docfx.json,
                 // user should not define it in file metadata
-                if (configMonikers.Length == 0)
+                if (configedMonikers.Length == 0)
                 {
-                    return (Errors.Versioning.MonikerRangeUndefined(metadata.MonikerRange), configMonikers);
+                    errors.Add(Errors.Versioning.MonikerRangeUndefined(metadata.MonikerRange));
+                    return (errors, configedMonikers);
                 }
 
                 var fileMonikers = _rangeParser.Parse(metadata.MonikerRange);
-                var intersection = configMonikers.Intersect(fileMonikers).ToArray();
+
+                // monikerRagen takes precedence over monikers since it is more likely from user configuration
+                if (fileMonikers.Length > 0 && metadata.Monikers.Length > 0)
+                {
+                    errors.Add(Errors.Versioning.DuplciateMonikerConfig(new SourceInfo(file, 0, 0)));
+                }
+                if (fileMonikers.Length == 0 && metadata.Monikers.Length > 0)
+                {
+                    fileMonikers = metadata.Monikers.Select(x => x.Value).ToArray();
+                }
+
+                var intersection = configedMonikers.Intersect(fileMonikers).ToArray();
 
                 // With non-empty config monikers,
                 // warn if no intersection of config monikers and file monikers
                 if (intersection.Length == 0)
                 {
-                    var error = Errors.Versioning.MonikeRangeOutOfScope(configMonikerRange, configMonikers, metadata.MonikerRange, fileMonikers);
-                    return (error, intersection);
+                    errors.Add(Errors.Versioning.MonikeRangeOutOfScope(configMonikerRange, configedMonikers, fileMonikers, metadata.MonikerRange, metadata.Monikers));
```

Comment:
nit: `fileMonikers` already is `metadata.Monikers`
```suggestion
                    errors.Add(Errors.Versioning.MonikeRangeOutOfScope(configMonikerRange, configedMonikers, fileMonikers, metadata.MonikerRange));
```

Reply:
Not really since I choose `MonikerRange` to take precedence over `Monikers`